### ISSUE TITLE
pathfinding modifiers: path generation written in C

### ISSUE THREAD

**Issue by rfavour** [frc2879/2017-newcomen#25]
to fix I will write the modifiers in C as well

**Comment by floogulinc** [comment]
Don't know what you mean by this

**Comment by rfavour** [comment]
the pathfinder library uses native to call code written in C, and the core of the path generation is not in the java version

**Comment by floogulinc** [comment]
Yeah but we should still be able to write our modifier in Java, it just wouldn't be as efficient.

**Comment by rfavour** [comment]
I suppose, but we will need to include the C code as well in the final version, and it isn't terribly complicated

**Comment by floogulinc** [comment]
No I don't think we will

**Comment by rfavour** [comment]
will there be no on the fly path generation?

**Comment by floogulinc** [comment]
I'll explain on Tuesday
